src/rlookup.c

```diff
@@ -890,13 +903,22 @@ static void RLookup_HGETALL_scan_callback(RedisModuleKey *key, RedisModuleString
   RLookup_HGETALL_privdata *pd = privdata;
   size_t fieldCStrLen;
   const char *fieldCStr = RedisModule_StringPtrLen(field, &fieldCStrLen);
-  RLookupKey *rlk = RLookup_GetKeyEx_TEMP(pd->it, fieldCStr, fieldCStrLen, RLOOKUP_F_OCREAT | RLOOKUP_F_NAMEALLOC);
-  if (!pd->options->noSortables && (rlk->flags & RLOOKUP_F_SVSRC)) {
-    return;  // Can load it from the sort vector on demand.
+  RLookupKey *rlk = RLookup_FindKey(pd->it, fieldCStr, fieldCStrLen);
+  if (!rlk) {
+    // First returned document, create the key.
+    uint32_t flags = pd->options->noSortables ? RLOOKUP_F_NAMEALLOC | RLOOKUP_F_FORCE_LOAD : RLOOKUP_F_NAMEALLOC;
+    rlk = RLookup_GetKey_LoadEx(pd->it, fieldCStr, fieldCStrLen, fieldCStr, flags);
+    if (!rlk) {
+      return; // Key is sortable, can load it from the sort vector on demand.
```

Comment:
Can we log assert this?

Sorting Vector will necessarrily have __all__ fields? (not related to this PR since it is not changed here)

Reply:
This is actually a dead code as of today, as the only one that call this function is the loader, and it always has the `noSortables` option.
This could be related to the old approach and to “force load” and override sortable. We might want to give up on this as we can now know Mitch better if the value is sortable, val_available or even loaded be an earlier loader. We can keep the `noSortables` option for json with dialect<3

Reply:
And this is not something to log. It just means some loader already loaded this specific field, or it’s val_available and `noSortables` is off